publication-rest/src/main/java/no/unit/nva/publication/update/PublishingRequestResolver.java

```diff
@@ -88,18 +112,21 @@ private boolean thereAreNoPendingFiles(Publication publicationUpdate) {
                    .noneMatch(PendingFile.class::isInstance);
     }
 
-    private List<PublishingRequestCase> fetchPendingPublishingRequestsForUserInstitution(
+    private List<FilesApprovalEntry> fetchPendingFileApprovalEntryForUserInstitutionOrWithFilesForApprovalWhenDegree(
         Publication publication) {
         return resourceService
                    .fetchAllTicketsForResource(Resource.fromPublication(publication))
-                   .filter(PublishingRequestCase.class::isInstance)
-                   .map(PublishingRequestCase.class::cast)
-                   .filter(ticketEntry -> ticketEntry.hasSameOwnerAffiliationAs(userInstance))
+                   .filter(FilesApprovalEntry.class::isInstance)
+                   .map(FilesApprovalEntry.class::cast)
+                   .filter(ticketEntry -> Resource.fromPublication(publication).isDegree()
+                                              ? !ticketEntry.getFilesForApproval().isEmpty()
+                                              : ticketEntry.hasSameOwnerAffiliationAs(userInstance))
```

Comment:
Is this just named method containing `(isDegree() && !ticketEntry.isEmpty()) || ticketEntry.hasSameOwnerAffiliationAs(userInstance)`?